makefile

```diff
@@ -29,3 +29,9 @@ total_lines:
 
 total_words:
 	wc -w *.txt
+
+setup:
+	python3 -m venv env
+	pip install --upgrade pip
```

Comment:
This won't actually work.  One of the nuances of makefiles is that the lines in each of the makefiles is a shell onto it's own.  In other words, 

* `python3 -m venv` creates a virtual environment in a directory called `env`
* `pip install --upgrade pip` executes pip in the outer environment, because the environment is not activated.

This won't work either...
* `python3 -m venv` creates a virtual environment in a directory called `env`
* `source env/bin/activate` activate the virtual environment
* `pip install --upgrade pip`, doesn't work because line 2 activated the env, then LEFT the shell for that line.

This will work for the second line...
* `source env/bin/activate; pip install --upgrade pip`

And it works because it's all in one line.  You enter the shell at the beggining of the line, activate the env, use pip in that env, and then at the end of the line you exit the shell and are back at the external environment.